fairscale/nn/checkpoint/checkpoint_activations.py

```diff
@@ -277,6 +277,7 @@ def forward(  # type: ignore
         with torch.no_grad(), enable_checkpointing():
             unpacked_args, unpacked_kwargs = unpack_kwargs(kwarg_keys, args)
             outputs = run_function(*unpacked_args, **unpacked_kwargs)
+            the_module = unpacked_args[0]
```

Comment:
what is this for?

Reply:
I think you may have already figured this out. But Darryl's commit removed the need of the_module but Alex's commit actually added a need for it concurrently. It is a rare occurrence of race condition and git merge didn't catch it, which is a really cool example of how tricky is software engineering. :-)

Reply:
I did not get a chance to see Darryl's PR yet but I did see Alex's commit. Thats a tricky race condition though!